jujugui/static/gui/src/app/state/state.js

```diff
@@ -0,0 +1,283 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+if (typeof this.jujugui === 'undefined') {
+  this.jujugui = {};
+}
+
+/** Class representing the State of the Juju GUI */
+const State = class State {
+  /**
+    Create a new instance of the GUI state.
+    @param {Object} cfg - The configuration options for a new State instance.
+  */
+  constructor(cfg) {
+    /**
+      The baseURL to use when generating URLs.
+      @type {String}
+    */
+    this.baseURL = cfg.baseURL;
+    /**
+      The list of dispatchers for the various components from the url.
+      @type {Object}
+    */
+    this.dispatchers = cfg.dispatchers;
+
+    if (cfg.seriesList && !Array.isArray(cfg.seriesList)) {
+      throw 'Series list must be an Array';
+    }
+    /**
+      The list of possible distro 'series' for the store paths. ex)
+      trusty, precise, xenial. The GUI has a utils method `getSeriesList` that
+      is the central storage location for the series information. This Object
+      should be converted to an array and passed to this state when being
+      instantiated.
+      @type {Array}
+    */
+    this.seriesList = (function() {
+      if (!cfg.seriesList) {
+        console.log('No series list provided, using default list');
+        return ['bundle', 'precise', 'trusty', 'xenial'];
```

Comment:
So, I'd make the seriesList config non-optional, because I don;t think there is a point in running with a partial list, as it would always lead to unexpected errors and unrecognised store sections.